internal/services/compute/virtual_machine_run_command_resource.go

```diff
@@ -0,0 +1,702 @@
+						Sensitive:    true,
+						ValidateFunc: validation.IsUUID,
+						ConflictsWith: []string{
+							"error_blob_managed_identity.0.client_id",
+						},
+					},
+				},
+			},
+		},
+
+		"error_blob_uri": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.IsURLWithHTTPS,
+		},
+
+		"output_blob_managed_identity": {
+			Type:      pluginsdk.TypeList,
+			Optional:  true,
+			MaxItems:  1,
+			Sensitive: true,
+			RequiredWith: []string{
+				"output_blob_uri",
+			},
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"client_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						Sensitive:    true,
+						ValidateFunc: validation.StringIsNotEmpty,
+						ConflictsWith: []string{
+							"output_blob_managed_identity.0.object_id",
+						},
+					},
+					"object_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						Sensitive:    true,
+						ValidateFunc: validation.StringIsNotEmpty,
+						ConflictsWith: []string{
+							"output_blob_managed_identity.0.client_id",
+						},
+					},
+				},
+			},
+		},
+
+		"output_blob_uri": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.IsURLWithHTTPS,
+		},
+
+		"parameter": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"value": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"protected_parameter": {
+			Type:      pluginsdk.TypeList,
+			Optional:  true,
+			Sensitive: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						Sensitive:    true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"value": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						Sensitive:    true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"run_as_password": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Sensitive:    true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"run_as_user": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"timeout_in_seconds": {
+			Type:         pluginsdk.TypeInt,
+			Optional:     true,
+			ValidateFunc: validation.IntAtLeast(1),
+		},
+
+		"location": commonschema.Location(),
+
+		"tags": commonschema.Tags(),
+	}
+
+}
+
+func (r VirtualMachineRunCommandResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"instance_view": {
```

Comment:
What would users do with this output? The `output` field _perhaps_ could be useful, but if the exit code is non-zero, presumably the resource would be marked as failed (as is common in other tooling)?

Reply:
The resource will run the script in VM when Create/Update is triggered. And if script fails, it will return error but the instance still exists in Azure, with `non-zero` exit code and error messages inside the `instance_view` block. The problem is: if save the failed instance in Terraform state, the resource will be marked as `taint` thus lead to force replacement during next apply.
```
Terraform will perform the following actions:

  # azurerm_virtual_machine_run_command.example03 is tainted, so must be replaced
```
but the API has some issue when performing replacement on same ID,

Reply:
Added a receate test for this case where a wrong script was submitted and then be corrected.